lib/net/ssh/transport/session.rb

```diff
@@ -61,12 +61,12 @@ def initialize(host, options={})
       @bind_address = options[:bind_address] || nil
       @options = options
 
-      debug { "establishing connection to #{@host}:#{@port}" }
-
       @socket =
         if (factory = options[:proxy])
+          debug { "establishing connection to proxy" }
```

Comment:
If we wanted to be pedantic here we would say:

"establishing proxy connection for #{host}"

and then ensure the proxy connection we are opening has a similar level of debugging so that we would eventually see:

"establishing connection to baz.quux.org"

However, as it stands, this PR is a big improvement over what exists today.